moshi/src/main/java/com/squareup/moshi/JsonUtf8Reader.java

```diff
@@ -860,23 +876,31 @@ private String nextUnquotedValue() throws IOException {
   }
 
   private void skipQuotedValue(ByteString runTerminator) throws IOException {
+    skipQuotedValue(runTerminator, BLACKHOLE);
+  }
+
+  private void skipQuotedValue(ByteString runTerminator, Sink sink) throws IOException {
     while (true) {
       long index = source.indexOfElement(runTerminator);
       if (index == -1L) throw syntaxError("Unterminated string");
 
       if (buffer.getByte(index) == '\\') {
-        buffer.skip(index + 1);
+        sink.write(buffer, index + 1);
         readEscapeCharacter();
```

Comment:
Should `readEscapeCharacter` have `sink` passed in ala https://github.com/infotech-group/moshi/commit/1d052e71e163b99a6f4b233f951edf9b71f646f8#diff-ba52b67c0349f268e571d765eab69d95

Reply:
Not all of those tests appeared to pass when I tried them in #1219